extensions/flac/src/main/jni/flac_parser.cc

```diff
@@ -452,8 +453,21 @@ int64_t FLACParser::getSeekPosition(int64_t timeUs) {
   for (unsigned i = mSeekTable->num_points; i > 0; ) {
     i--;
     if (points[i].sample_number <= sample) {
-      return firstFrameOffset + points[i].stream_offset;
+      result[0] = points[i].sample_number / getSampleRate() * 1000000LL;
+      result[1] = firstFrameOffset + points[i].stream_offset;
+      if (i + 1 <= mSeekTable->num_points) {
```

Comment:
No need to push any further changes as we'll edit this during merge, but just to sanity check my thinking:

* This should be `<` rather than `<=`, right?
* I'm also going to adjust this to use the else path if `points[i].sample_number == sample` (i.e. the exact seek case). Does that make sense?

Reply:
Are you talking about the Line 455 ("if (points[i].sample_number <= sample) {") ?
If yes, that makes sense.

Reply:
I meant that this line should use `i + 1 < mSeekTable->num_points` rather than `i + 1 <= mSeekTable->num_points`.

The biggest index you can use into `points` is `num_points - 1` (due to 0-indexing), but the current logic allows `num_points` to be used. I was able to make the current implementation return a broken looking seek point by seeking right to the end of a stream as a result.